signal/signal.go

```diff
@@ -49,3 +51,10 @@ func (s *Signal) AddError(err error) {
 	}
 	s.Status = 1
 }
+
+func (s *Signal) PanicHandler(ctx context.Context) {
+	if r := recover(); r != nil {
+		svc1log.FromContext(ctx).Error(fmt.Sprintf("panic: %v", r))
+		s.AddError(fmt.Errorf("panic: %v", r))
```

Comment:
should we propagate the panic?

Reply:
No, the whole thing I'm trying to achieve here is that we do not fail the actual tool run but we exit cleanly with the error and status code captured in the Signal itself